nemo/collections/asr/parts/submodules/rnnt_malsd_batched_computer.py

```diff
@@ -0,0 +1,1125 @@
+            out=self.state.decoder_output,
+        )
+        self.decoder.batch_replace_states_mask(
+            src_states=self.state.prev_decoder_state,
+            dst_states=self.state.decoder_state,
+            mask=preserve_state.view(-1),
+            other_src_states=decoder_state,
+        )
+
+        if self.ngram_lm_batch is not None:
+            # batch_lm_states: size: [(batch_size x beam_size)]
+            # batch_lm_states_candidates: [(batch_size x beam_size) x V (without blank)]
+            self.state.batch_lm_states_candidates.copy_(
+                torch.gather(
+                    self.state.batch_lm_states_candidates,
+                    dim=1,
+                    index=self.state.next_idx[:, :, None].expand(
+                        self.state.batch_size, self.beam_size, self.state.batch_lm_states_candidates.shape[-1]
+                    ),
+                )
+            )
+            torch.gather(
+                self.state.batch_lm_states, dim=1, index=self.state.next_idx, out=self.state.batch_lm_states_prev
+            )
+            last_labels_wb_blank_replaced = torch.where(preserve_state, 0, self.state.last_labels_wb)
+
+            torch.gather(
+                self.state.batch_lm_states_candidates,
+                dim=-1,
+                index=last_labels_wb_blank_replaced.unsqueeze(-1),
+                out=self.state.batch_lm_states.unsqueeze(-1),
+            )
+            torch.where(
+                preserve_state,
+                self.state.batch_lm_states_prev,
+                self.state.batch_lm_states,
+                out=self.state.batch_lm_states,
+            )
+
+            lm_scores, batch_lm_states_candidates = self.ngram_lm_batch.advance(
+                states=self.state.batch_lm_states.view(-1)
+            )  # vocab_size_no_blank
+            lm_scores = (
+                lm_scores.to(dtype=self.state.float_dtype).view(self.state.batch_size, self.beam_size, -1)
+                * self.ngram_lm_alpha
+            )
+
+            self.state.batch_lm_states_candidates.copy_(
+                batch_lm_states_candidates.view(self.state.batch_size, self.state.beam_size, -1)
+            )
+            self.state.lm_scores.copy_(lm_scores)
+
+        # step 6: update time indices + active mask
+        self.state.time_indices.copy_(self.state.batched_hyps.next_timestamp)
+        torch.minimum(self.state.time_indices, self.state.last_timesteps, out=self.state.safe_time_indices)
+        torch.less_equal(self.state.time_indices, self.state.last_timesteps, out=self.state.active_mask)
+        torch.any(self.state.active_mask, out=self.state.active_mask_any)
+
+    def __call__(
+        self,
+        x: torch.Tensor,
+        out_len: torch.Tensor,
+    ) -> BatchedBeamHyps:
+        if self.cuda_graphs_mode is not None and x.device.type == "cuda":
+            return self.modified_alsd_cuda_graphs(encoder_output=x, encoder_output_length=out_len)
```

Comment:
Please, add here and for TDT: `with torch.amp.autocast(device_type="cuda", enabled=False):`

Similar to https://github.com/NVIDIA/NeMo/pull/12938